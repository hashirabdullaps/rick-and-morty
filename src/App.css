@import '~antd/dist/antd.css';

.spinner-loading {
  margin: 0 auto;
  text-align: center;
  width: 100vw;
  padding: 10px;
}
