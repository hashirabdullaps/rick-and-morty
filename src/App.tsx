import React from 'react';
import { Link, Route, Routes, Navigate } from "react-router-dom";
import { Button } from 'antd';

import './App.css';
import notFound from './assets/images/404.png'
import LayoutWrapper from './layout/Layout';
import Home from './pages/home/Home';
import CharacterDetailPage from './pages/character-detail/CharacterDetail';

function App() {
  return (
    <>
      <Routes>
        <Route path="/" element={<LayoutWrapper />}>
          <Route index element={<Home />} />
          <Route path="/character/:id" element={<CharacterDetailPage />} />
          <Route path="/404-Not-Found" element={<NoMatch />} />
          <Route
            path="*"
            element={<Navigate to="/404-Not-Found" replace />}
          />
        </Route>
      </Routes>
    </>
  );
}

export default App;

function NoMatch() {
  return (
    <div style={{ padding: '20px' }}>
      <img src={notFound} alt="404" />
      <Button type="primary">
        <Link to="/">Go to the home page</Link>
      </Button>
    </div>
  );
}
