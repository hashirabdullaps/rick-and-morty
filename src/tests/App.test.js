import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';
import { BrowserRouter } from "react-router-dom";
import { MockedProvider } from '@apollo/client/testing';

test('should render App', () => {
  const mocks = [];

  const component = renderer.create(
    <MockedProvider mocks={mocks} addTypename={false}>
      <BrowserRouter><App /></BrowserRouter>
    </MockedProvider>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});
