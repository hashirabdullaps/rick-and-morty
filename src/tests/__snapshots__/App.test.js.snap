// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App 1`] = `
<section
  className="ant-layout"
  style={
    Object {
      "minHeight": "100vh",
    }
  }
>
  <header
    className="ant-layout-header header"
  >
    <img
      alt="logo"
      className="header__logo"
      onClick={[Function]}
      src="logo.png"
    />
  </header>
  <main
    className="ant-layout-content"
  >
    <div
      className="characterList"
    >
      <h1>
        All Characters
      </h1>
      <div
        className="infinite-scroll-component__outerdiv"
        style={Object {}}
      >
        <div
          className="infinite-scroll-component characterList_scroll"
          style={
            Object {
              "WebkitOverflowScrolling": "touch",
              "height": "auto",
              "overflow": "auto",
            }
          }
        >
          <div
            className="card-grid"
          />
        </div>
      </div>
    </div>
  </main>
  <footer
    className="ant-layout-footer footer"
  >
    <a
      className="footer__link"
      href="/"
    >
      Rick and Morty by Hashir Â© 2022
    </a>
  </footer>
</section>
`;
